pkg/sources/git/git.go

```diff
@@ -532,37 +551,28 @@ func (s *Git) ScanCommits(ctx context.Context, repo *git.Repository, path string
 
 	gitDir := filepath.Join(path, gitDirName)
 
-	logger := repoCtx.Logger()
-	var logValues []any
-	if scanOptions.BaseHash != "" {
-		logValues = append(logValues, "base", scanOptions.BaseHash)
-	}
-	if scanOptions.HeadHash != "" {
-		logValues = append(logValues, "head", scanOptions.HeadHash)
-	}
 	logger.V(1).Info("scanning repo", logValues...)
 
 	var depth int64
 	var lastCommitHash string
 	for diff := range diffChan {
-		if scanOptions.MaxDepth > 0 && depth >= scanOptions.MaxDepth {
+		if maxDepthSet && depth >= scanOptions.MaxDepth {
 			logger.V(1).Info("reached max depth", "depth", depth)
 			break
 		}
 
 		fullHash := diff.Commit.Hash
-		if !strings.EqualFold(fullHash, lastCommitHash) {
+		if baseHashSet && scanOptions.BaseHash == fullHash {
+			logger.V(1).Info("reached base commit", "commit", fullHash)
+			break
+		}
+
+		if fullHash != lastCommitHash {
```

Comment:
Why aren't we worried about case sensitivity anymore?

Reply:
This was how it was initially implemented, I had changed it to a case insensitive version. I'm reverting it back to the original approach.